### DISCUSSION TITLE
json.exception.parse_error.101 causes core dumped (bad char in json::parse)

### DISCUSSION THREAD

**Discussion by Vlczech** [nlohmann/json#2467] [Q&A] [answered]
#### What is the issue you have?

**code**
```
// Example program
#include <iostream>
#include <string>

int main()
{
  std::string tmp(1, '\x1b');
  json::parse(tmp);
}

```

**result**
```
terminate called after throwing an instance of 'nlohmann::detail::parse_error'
  what():  [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - invalid literal; last read: '<U+001B>'
Aborted (core dumped)
```

#### Please describe the steps to reproduce the issue.

_...as above_

#### Can you provide a small but working code example?

_...as above_

#### What is the expected behavior?

I think, that it could be exited by error but not core dump

#### And what is the actual behavior instead?
aborted (core dumped) _as abore_

#### Which compiler and operating system are you using?

- Compiler: g++ (Ubuntu 9.3.0-10ubuntu2) 9.3.0
- Operating system: ubuntu:rolling in docker (Linux ee2fa91ec903 5.7.19-desktop-1.mga7 #1 SMP Thu Aug 27 20:27:55 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux)

#### Which version of the library did you use?

- [ x ] latest release version 3.9.1
- [ ] other release - please state the version: ___
- [ ] the `develop` branch

#### If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?

- [ ] yes
- [ x ] no - please copy/paste the error message below

**Comment by nlohmann** [comment, marked as answer]
Your input is invalid JSON, and with default parameters, this yields an exception. As this is not caught in your code, your program terminates. You can fix this by adding a `try`/`catch (json::exception &e) { std::cerr << e.what() << std::endl; }`.

**Reply by Vlczech** [reply]
@nlohmann: Yes i know, I added catch. But I was surprised that without catch it was core dumped and not exception only without core dumped - that's why I report it.

**Reply by nlohmann** [reply]
But that’s not related to the library, but just how C++ works.
